src/EnergyPlus/HeatBalanceHAMTManager.hh

```diff
@@ -112,14 +115,65 @@ namespace HeatBalanceHAMTManager {
 
         // Default Constructor
         subcell()
-            : matid(-1), sid(-1), Qadds(0.0), density(-1.0), wthermalc(0.0), spech(0.0), htc(-1.0), vtc(-1.0), mu(-1.0), volume(0.0), temp(0.0),
-              tempp1(0.0), tempp2(0.0), wreport(0.0), water(0.0), vp(0.0), vpp1(0.0), vpsat(0.0), rh(0.1), rhp1(0.1), rhp2(0.1), rhp(10.0),
-              dwdphi(-1.0), dw(-1.0), origin(3, 0.0), length(3, 0.0), overlap(6, 0.0), dist(6, 0.0), adjs(6, 0), adjsl(6, 0)
+            : matid(-1), sid(-1), layer_id(-1), source_id(-1), Qadds(0.0), Wadds(0.0), density(-1.0), wthermalc(0.0), spech(0.0), htc(-1.0),
+              vtc(-1.0), mu(-1.0), volume(0.0), temp(0.0), tempp1(0.0), tempp2(0.0), wreport(0.0), water(0.0), vp(0.0), vpp1(0.0), vpsat(0.0),
+              rh(0.1), rhp1(0.1), rhp2(0.1), rhp(10.0), dwdphi(-1.0), dw(-1.0), origin(3, 0.0), length(3, 0.0), overlap(6, 0.0), dist(6, 0.0),
+              adjs(6, 0), adjsl(6, 0)
         {
         }
     };
 
-    void ManageHeatBalHAMT(EnergyPlusData &state, int const SurfNum, Real64 &SurfTempInTmp, Real64 &TempSurfOutTmp);
+    // structure internal moisture source
+    struct InternalMoistureSource
+    {
+        int id;         // Internal moisture source id
+        int surface_id; // Surface id
+        int layer_id;   // Layer id
+
+        // int SourceType; // (1 = air flow input; 2 = infiltration model (Kuenzel); 3 = multi zone air flow model)
+        enum class Type
+        {
+            UserDefined = 1,
+            StackAndOverPressure,
+            AirflowNetwork
```

Comment:
Should have Invalid = -1 and Num as the last element ? Any reason UserDefinded must = 1 ?